app/servers/main/js/favorites.js

```diff
@@ -29,7 +30,17 @@ var favorites = {
   },
 
   _save: function() {
-    asyncStorage.setItem(this.KEYNAME, this.items);
+    var deferred = Promises.defer();
+    asyncStorage.setItem(
+      this.KEYNAME,
+      this.items,
+      () => deferred.resolve(),
+      (e) => {
```

Comment:
I don't reject with e directly because I cannot pass it to postMessage (DataCloneError).